src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -668,6 +674,43 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
                         "Could not install root log handler on node: " + c.getName(), e);
             }
         }
+
+        @Override
+        public void onOffline(@Nonnull final Computer c, @CheckForNull OfflineCause cause) {
+            if (Boolean.getBoolean(SupportPlugin.class.getName() + ".generateOnOfflineNode")
+                    && cause != null && !cause.getClass().getName().contains("RestartCause")) {
+                Timer.get().schedule(new java.util.concurrent.Callable<Void>() {
+                    @Override
+                    public Void call() throws Exception {
+                        boolean locked = false;
+                        try {
+                            locked = lock.tryLock();
```

Comment:
Should this lock be shared with `PeriodicWorkImpl` so we do not try to generate two background bundles at once?